sdcm/nemesis.py

```diff
@@ -4904,6 +4905,121 @@ def are_gate_closed_messages_raised(log_reader):
+
+        @contextlib.contextmanager
+        def use_send_signal_process(target_node: BaseNode):
+            target_node.log.debug("Send signal SIGSTOP to scylla process on node %s", target_node.name)
+            target_node.remoter.sudo("pkill --signal SIGSTOP -e scylla", timeout=60)
+            yield
+            target_node.log.debug("Send signal SIGCONT to scylla process on node %s", target_node.name)
+            target_node.remoter.sudo(cmd="pkill --signal SIGCONT -e scylla", timeout=60)
+
+        simulate_node_unavailability = use_network_firwall if use_iptables else use_send_signal_process
+
+        working_node = random.choice([node for node in self.cluster.nodes if node != self.target_node])
+        target_host_id = self.target_node.host_id
+
+        def check_node_removed_from_cluster(node: BaseNode, verification_node: Optional[BaseNode] = None):
+            cluster_status: Optional[dict] = self.cluster.get_nodetool_status(verification_node=verification_node)
+            if not cluster_status:
+                return False
+            result = []
+            for dc in cluster_status:
+                result.append(node.ip_address not in list(cluster_status[dc].keys()))
+            return all(result)
+
+        def check_node_is_down_in_cluster(down_node: BaseNode, verification_node: BaseNode):
+            return down_node not in self.cluster.get_nodes_up_and_normal(verification_node)
+
+        prepare_test_keyspace(self.target_node)
+        with simulate_node_unavailability(self.target_node):
+            # target node stopped by Contextmanger. Wait while its status will be updated
+            wait_for(check_node_is_down_in_cluster, timeout=180, throw_exc=False,
+                     down_node=self.target_node, verification_node=working_node)
+            self.log.debug("Remove node %s : hostid: %s with stopped scylla from cluster",
+                           self.target_node.name, target_host_id)
+            working_node.run_nodetool(f"removenode {target_host_id}", retry=3)
+            self.log.debug("Wait while all node update topology status")
+            wait_for(check_node_removed_from_cluster, timeout=300, throw_exc=False,
+                     node=self.target_node, verification_node=working_node)
+            # Run read barrier on working node to validate that it update raft topology state
+            read_barrier(working_node)
+
+        # Context manager at exit  start scylla on target node.
+        # But node already removed from cluster. So any operations from it
+        # should be banned. If query executed succesfull, raise an error
+        try:
+            with self.cluster.cql_connection_exclusive(node=self.target_node) as session:
+                stmt = SimpleStatement("INSERT INTO banned_keyspace.table1 (id, name) VALUES (1, 'name1');",
+                                       consistency_level=ConsistencyLevel.ALL)
```

Comment:
shouldn't we test also consistency level `ONE`?
so later we could validate if inserted row didn't go through some other path than query.

Reply:
Level One, could be unstable, because sometime the query will succesfully write on banned node (if partition will belong to the node) and some time it will be failed, because the query will be executed on the the not available nodes

Reply:
@aleksbykov IIUC what you're trying to do here, it will work deterministically only when cluster size is 3.
Since you're using RF=3, it doesn't force one replica to reside on the target node that you just took out of the cluster.
Maybe you should change the RF of this Banned keyspace to be as big as the cluster size.

Reply:
Going back to this, with bit more understanding of what you're trying to test here.
CL=ALL is enough here as long as RF > 1, since it tries to replicate the other replicas to a cluster that it's not part of anymore.